src/Game.java

```diff
@@ -1,45 +1,119 @@
+
+    for (int i = 0; i < numPlayers; i++) {
+      System.out.println("Please enter the name of player " + (i + 1) + ":");
+      String playerName;
+      while (true) {
+        playerName = scanner.nextLine();
+        if (!allPlayers.stream().map(Object::toString)
+            .collect(Collectors.toList()).contains(playerName)) {
+          if (playerName.length() != 0) {
+            break;
+          }
+          System.out.println("Invalid name, please enter another name");
+        } else {
+          System.out.println("Please enter another name, " + playerName
+              + " has already been taken:");
         }
+      }
 
-        int currentLeaderIndex = 0;
+      allPlayers.add(new Player(playerName));
+    }
+
+    int currentLeaderIndex = 0;
+
+    mainLoop:
+    while (true) {
 
-        while (true) {
-            System.out.println("Do you wish to start a new game? Y/N");
-            String answer = scanner.nextLine();
+      Player leader = allPlayers.get(currentLeaderIndex);
 
-            if (answer.startsWith("N")) {
-                break;
-            }
+      Word word;
+      System.out.println(leader + ", you are the leader, please pick a word:");
 
-            Player leader = allPlayers.get(currentLeaderIndex);
+      while (true) {
+        word = new Word(scanner.nextLine());
+        if (word.wordLength() != 0) {
+          break;
+        }
+        System.out.println("Please enter a word with at least 1 letter:");
+      }
+
+      List<Player> players = new ArrayList<>(allPlayers);
+      players.remove(leader);
 
-            System.out.println(leader + ", you are the leader, please pick a word:");
+      Session session = new Session(scanner, players, leader, word);
+      session.play();
 
-            Word word = new Word(scanner.nextLine());
+      currentLeaderIndex = (currentLeaderIndex + 1) % allPlayers.size();
+      System.out.println("Current Standings: ");
+      System.out.println(leaderboardToSet(allPlayers));
 
-            List<Player> players = new ArrayList<>(allPlayers);
-            players.remove(leader);
+      System.out.println("Do you wish to start a new game? Y/N");
+      String answer = scanner.nextLine();
 
-            Session session = new Session(scanner, players, leader, word);
-            session.play();
+      while (!answer.startsWith("y") && !answer.startsWith("Y")) {
+        if (answer.startsWith("N") || answer.startsWith("n")) {
+          break mainLoop;
+        } else {
+          System.out.println("Sorry, couldn't understand");
+          System.out.println("Do you wish to start a new game? Y/N");
+          answer = scanner.nextLine();
+        }
+      }
 
-            currentLeaderIndex = (currentLeaderIndex + 1) % allPlayers.size();
+      System.out.println("Thank you for playing! These are our top players:");
+      int topScore = allPlayers.get(0).getScore();
+      List<Player> topPlayers = new ArrayList<>();
+      for (Player p : allPlayers) {
```

Comment:
This doesn't really make any sense. You initialise topScore with the score of the first player, and for each player that has a greater score than that, you add them to topPlayers. That means topPlayers depends on the order of allPlayers which is completely arbitrary. I.e. if the first player in the allPlayers list has the highest score, he is gonna be the only one added to topPlayers.

You want to probably sort your leaderboard here and display all names (or truncate the top 3, or whatever).